### PR TITLE
#6821 feat: ImageBanner component

### PR DESCRIPTION
**Pull request by creido-welly**
Work in progress ...

### CODE REVIEW

File: src/components/ImageBanner/ImageBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  src,
+  srcSet
+}: ImageBannerProps) => {
+  const classNames = cx('cc-image-banner', {
+    [`${className}`]: className
+  });
+
+  return (
+    <figure className={classNames}>
+      <div className="cc-image-banner__container" role="banner">
```
Comment (lukewhitmore) [inline src/components/ImageBanner/ImageBanner.tsx]: Could this div and the previous figure be combined to simplify?

File: src/components/ImageBanner/ImageBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <ImageElement
+          alt={alt}
+          className="cc-image-banner__img"
+          src={src}
+          role="presentation"
+          sizes={sizes}
+        />
+      </div>
+
+      <div className="cc-image-banner__caption-container">
```
Comment (lukewhitmore) [inline src/components/ImageBanner/ImageBanner.tsx:42]: Should this div be a figcaption element?

File: src/components/ImageBanner/ImageBanner.tsx (hunk 1 above)
Comment (lukewhitmore) [reply src/components/ImageBanner/ImageBanner.tsx]: If they can't be simplified .. perhaps ARIA role should be on figure anyway? Might make more sense?

File: src/components/ImageBanner/ImageBanner.tsx (hunk 2 above)
Comment (creido-welly) [reply src/components/ImageBanner/ImageBanner.tsx:42]: Good spot, MediaCaption component which this wraps is actually a figcaption

File: src/components/ImageBanner/ImageBanner.tsx (hunk 1 above)
Comment (creido-welly) [reply src/components/ImageBanner/ImageBanner.tsx]: It looks like they could be just looking at it briefly so I've made this change now.

This is largely a copy and paste from Drupal 7 right now as it's WIP ... more to follow

**Review by lukewhitmore — APPROVED** [review]

**Pull request merged**